### DISCUSSION TITLE
Serverports installation failed in Ubuntu

### DISCUSSION THREAD

**Discussion by voidFunctionReturn0** [serverpod/serverpod#812] [Q&A] [answered]
I am having problems installing serverpod on ubuntu 22.04.

First, I installed flutter and docker.

Next, I ran the command below to install the server port.
"$ dart pub global activate serverpod_cli"

However, while the installation is in progress, the message 'Building package executables' is displayed and the installation does not finish even after more than 83 minutes.

I am wondering how to solve this.

![서버포트 설치 에러](https://user-images.githubusercontent.com/17719773/230903038-cc045c6b-e6a1-499d-a66c-82e3d5087221.png)

**Comment by yahu1031** [comment, marked as answer]
Make sure that has enough swap memory allocated. @voidFunctionReturn0

**Reply by voidFunctionReturn0** [reply]
Thank you!

**Reply by yahu1031** [reply]
worked?

**Reply by voidFunctionReturn0** [reply]
> worked?

Thanks to you, I solved the problem by adding swap memory.
However, even after that, the execution of the command to check the serverpod installation failed, and the app made with the serverpod also ran into a problem that did not run.

![serverpod error3](https://user-images.githubusercontent.com/17719773/231898345-8b585a46-497c-4649-921b-bf6922787114.png)

**Reply by yahu1031** [reply]
Check if the path exists and check if it is hard coded somewhere. Looks like the path was hard coded somewhere.

**Reply by voidFunctionReturn0** [reply]
(Solved)

Thank you very much for your help. It was my fault for not reading the warning message carefully. Thanks to you, I succeeded in running the serverpod command.

But unfortunately, the problem occurred again when running the serverpod app.

![serverpod-error](https://user-images.githubusercontent.com/17719773/232178167-0fbd8d9e-e05e-485a-bc1d-ad6e186fc4fa.png)


My development environment is windows,
and my deployment environment is ubuntu.


-----------------------


I solved this problem.

The following command was executed in the server project root path.
$flutter packages upgrade
